src/main/java/core/basesyntax/figures/figurehandlers/FigureSupplier.java

```diff
@@ -0,0 +1,60 @@
+package core.basesyntax.figures.figurehandlers;
+
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.Figure;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURES_COUNT = 4; //from 0 to 4
+    private final Random random = new Random();
+
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int randomIndex = random.nextInt(FIGURES_COUNT) + 1;
+
+        switch (randomIndex) {
+            case 1: {
+                String color = colorSupplier.getRandomColor();
+                int radius = random.nextInt(30) + 1;
+                return new Circle(color, radius);
+            }
+            case 2: {
+                String color = colorSupplier.getRandomColor();
+                int lowBase = random.nextInt(30) + 1;
+                int highBase = random.nextInt(30) + 31;
+                int leg = random.nextInt(20) + 1;
+                int height = random.nextInt(10) + 1;
```

Comment:
you have way to many magic numbers in each case

i would suggest not to care that much about random values for each figure property but rather create a simple method that would return a random number, bounded by some max number (for example `100`) and use it for all properties at the same time